lib/enum/basic_enum.rb

```diff
@@ -85,66 +97,79 @@ def element_frequency_count
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.find_all { |x| x.end_with? 'er'}
   end
 
 
   # Refer to find_index
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-index
   def index_of_first_awesome_element
+    collection.find_index { |name| name[:awesome] == true}
   end
 
   # Refer to group_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-group-by
   def group_elements_by_favorite_language
+    collection.group_by { |people| people[:favorite_language]}
```

Comment:
And here, more accurate to use `|person|` instead of `|people|`, since group_by goes through the collection one person at a time.